### ISSUE TITLE
Try to maintain `Fixes #...` messages from squashed commits?

### ISSUE THREAD

**Issue by bnjbvr** [EmbarkStudios/octobors#20]
When working a pull request, developers may embed `Fixes #...` annotations within commit messages. Right now it seems that when octobors squashes commits, it uses the pull request's description as the commit message. As a result this will lose the annotations from the commit messages by default. It would be nice if it had the ability to parse the commit messages from the pull request, and keep these annotations in the final squashed commit's message.

**Comment by repi** [comment]
We prioritised getting the PR description for the squash commits as that is the main thing one has when the branch has multiple commits.

Could be possible for single-commit branches/merges to somehow combine the PR description with the commit description, or just use the commit description - but we often had more detailed PR descriptions than the commit descriptions so does sound a bit tricky and that one may have to just pick either or.

So the PR should have the "Fixes XX" strings also right, as otherwise the PR won't be linked and close the issues it is fixing, are you saying here that Octobers removes those lines from the PR description?

**Comment by bnjbvr** [comment]
> So the PR should have the "Fixes XX" strings also right, as otherwise the PR won't be linked and close the issues it is fixing, are you saying here that Octobers removes those lines from the PR description?

Not quite. In my case, the PR didn't have the `Fixes XX` string, but one commit in the PR contained it in its commit message. The commit messages are dropped when squashing, and Github doesn't propagate the `Fixes XX` annotations from the commit messages to the pull request's description automatically, so the `Fixes XX` was lost when squashing and merging.

**Comment by repi** [comment]
Seems to partially work, those commits in the PR didn't make the PR list the issues that it will resolve in the linked issues list:

<img width="303" alt="image" src="https://user-images.githubusercontent.com/1262692/117820597-fadd0b80-b26a-11eb-8f0d-907e3799418c.png">

So do think it would be safer to rely on mentioning issues in the PR description directly still.
